packages/orchestrator/internal/template/server/create_template.go

```
@@ -78,11 +76,12 @@ func (s *ServerStore) TemplateCreate(ctx context.Context, templateRequest *templ
 		)
 		defer buildSpan.End()
 
-		res, err := s.builder.Build(buildContext, template)
-		// Wait for the CLI to load all the logs
-		// This is a temporary ~fix for the CLI to load most of the logs before finishing the template build
-		// Ideally we should wait in the CLI for the last log message
-		time.Sleep(8 * time.Second)
+		externalWriter := writer.New(logger)
+		defer externalWriter.Close()
+
+		logsWriter := io.MultiWriter(&logs, externalWriter)
+		res, err := s.builder.Build(buildContext, template, logsWriter)
+		externalWriter.Sync()
```

if you want to ignore the error, make it explicit

---

I need to sync the logs before reporting a status because when the CLI will fetch status of the build (with logs), and it will see that the build has finished, no new request for logs will happen. At that point, the logs need to be complete.  If they weren't complete, missing lines would never be fetched again.

---

but logs from external writer goes through vector to loki, so you can't really coordinate it only with this, right?

---

Hmm, thats true, removing the sync